### PR TITLE
Upgrade to typescript-1.4

### PR DESCRIPTION
**Pull request by bluong**
Run
`npm update && npm install`
to update your typescript.

We should regression pass this as this is a library/language upgrade?

### CODE REVIEW

File: plottable.d.ts [hunk 1]
```diff
@@ -113,9 +113,9 @@ declare module Plottable {
              * @returns {[{ [key: string]: any }} coppied map.
              */
             function copyMap<T>(oldMap: {
-                [x: string]: T;
+                [key: string]: T;
             }): {
-                [x: string]: T;
+                [key: string]: T;
```
Comment (jtlan) [inline plottable.d.ts]: ?? I'm bewildered that TS1.3 didn't do this correctly.

File: plottable.d.ts [hunk 2]
```diff
@@ -3745,7 +3745,7 @@ declare module Plottable {
              */
             _doHover(p: Point): HoverData;
         }
-        class Hover extends AbstractInteraction {
+        class Hover extends Interaction.AbstractInteraction {
```
Comment (jtlan) [inline plottable.d.ts]: Why does this only happen here and not for `Click extends AbstractInteraction`?

**Comment by jtlan** [comment]
No real problems, just confused by inconsistencies that the upgrade seems to have introduced.

Also, no JS code was changed, so we can probably NOQE this?

File: plottable.d.ts (hunk 2 above)
Comment (bluong) [inline plottable.d.ts]: Weird.  This is really inconsistent?

**Comment by bluong** [comment]
Oh didn't notice that no JS code has changed.  Yep should be NOQE

**Pull request merged**